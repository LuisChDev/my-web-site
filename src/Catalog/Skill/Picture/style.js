import styled, {css, keyframes} from 'styled-components';
import posed from 'react-pose';
import {cubeColors, cubeImages} from '../../../style.js';
import React from 'react';

/**
 * wrapper for the cube perspective.
 */
const CubeWrapper = styled.div`
  margin-top: 10vh;
  perspective: 1000px;
  perspective-origin: 50% 50%;
`;

/**
 * animation, how the cube should look
 */
const cubeMotion = keyframes`
  from {
    transform: rotateX(0deg) rotateY(0deg);
  }

  to {
    transform: rotateX(360deg) rotateY(360deg);
  }
`;

/**
 * styles for the cube.
 */
const PosedCube = styled.div`
  margin: auto;
  position: relative;
  height: 24vw;
  width: 24vw;
  transform-style: preserve-3d;
  animation: ${cubeMotion} 20s linear infinite;
`;
// const StyledCubeWrap = ({hostRef}) => <StyledCube ref={hostRef}/>;


// /**
//  * container div for the whole cube.
//  */
// const PosedCube = posed(StyledCubeWrap)({
//   rotating: {
//     rotateX: ({rotatex}) => rotatex,
//     rotateY: ({rotatey}) => rotatey,
//   }
// });

/**
 * each individual face of the cube.
 */
const StyledFace = styled.div`
  position: absolute;
  box-sizing: border-box;
  padding: 10px;
  height: 100%;
  width: 100%;
  opacity: 0.8;
  border: solid 1px #eeeeee;
  ${({logo}) => css`
    background-image: url(${cubeImages[logo]});
  `};
  background-size: 150px 150px;
  background-repeat: no-repeat;
  background-position: center;
  ${props => css`
    transform: translate${props.t}(${props.td}vw) rotate${props.r}(${props.rd}deg);
    background-color: ${cubeColors[props.i]};
  `}
`;

export {PosedCube, StyledFace, CubeWrapper};
