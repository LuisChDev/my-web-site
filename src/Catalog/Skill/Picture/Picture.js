import React, {Component} from 'react';
import {StyledCube, StyledFace, CubeWrapper, CubeImg} from './style.js';

/**
 * this is the component that will be animated with CSS.
 */
const Picture = ({logo, pose}) => {
    return(
      <CubeWrapper>
        <StyledCube pose={pose}>
          {[["X", 0, "Z", 8],
            ["X", 180, "Z", -8],
            ["Y", 90, "X", 8],
            ["Y", -90, "X", -8],
            ["X", 90, "Y", 8],
            ["X", -90, "Y", -8]].map((x,i) =>
                                      <StyledFace r={x[0]} rd={x[1]}
                                                  t={x[2]} td={x[3]}
                                                  i={i} logo={logo}
                                                  key={i}/>)}
        </StyledCube>
      </CubeWrapper>
    );
};

export default Picture;
