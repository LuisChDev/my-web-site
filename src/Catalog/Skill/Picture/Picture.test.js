import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import Picture from './Picture.js';

describe("Picture component", () => {
  let output;
  beforeEach(() => {
    output = shallow(
      <Picture
        logo="testLogo"
        pose="testPose"/>
    );
  });

  it("renders correctly", () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
