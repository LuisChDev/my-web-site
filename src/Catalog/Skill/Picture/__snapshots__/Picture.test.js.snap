// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture component renders correctly 1`] = `
<styled.div>
  <Styled(Component)
    pose="testPose"
  >
    <styled.div
      i={0}
      key="0"
      logo="testLogo"
      r="X"
      rd={0}
      t="Z"
      td={8}
    />
    <styled.div
      i={1}
      key="1"
      logo="testLogo"
      r="X"
      rd={180}
      t="Z"
      td={-8}
    />
    <styled.div
      i={2}
      key="2"
      logo="testLogo"
      r="Y"
      rd={90}
      t="X"
      td={8}
    />
    <styled.div
      i={3}
      key="3"
      logo="testLogo"
      r="Y"
      rd={-90}
      t="X"
      td={-8}
    />
    <styled.div
      i={4}
      key="4"
      logo="testLogo"
      r="X"
      rd={90}
      t="Y"
      td={8}
    />
    <styled.div
      i={5}
      key="5"
      logo="testLogo"
      r="X"
      rd={-90}
      t="Y"
      td={-8}
    />
  </Styled(Component)>
</styled.div>
`;
